pkg/es/config/config.go

```diff
@@ -238,20 +248,20 @@ func (c *Configuration) ApplyDefaults(source *Configuration) {
 	if c.AdaptiveSamplingLookback == 0 {
 		c.AdaptiveSamplingLookback = source.AdaptiveSamplingLookback
 	}
-	if c.NumShards == 0 {
-		c.NumShards = source.NumShards
+	if c.Indices.Dependencies.TemplateOptions.NumShards == 0 {
+		c.Indices.Dependencies.TemplateOptions.NumShards = source.Indices.Dependencies.TemplateOptions.NumShards
 	}
-	if c.NumReplicas == 0 {
-		c.NumReplicas = source.NumReplicas
+	if c.Indices.Dependencies.TemplateOptions.NumReplicas == 0 {
+		c.Indices.Dependencies.TemplateOptions.NumReplicas = source.Indices.Dependencies.TemplateOptions.NumReplicas
 	}
-	if c.PrioritySpanTemplate == 0 {
-		c.PrioritySpanTemplate = source.PrioritySpanTemplate
+	if c.Indices.Spans.TemplateOptions.Priority == 0 {
```

Comment:
instead of repeating this for each index type, I would define a helper function and call it for each index